### ISSUE TITLE
pyjks does not exclude its `tests` directory when packaged

### ISSUE THREAD

**Issue by aetherknight** [kurtbrose/pyjks#35]
When I install pyjks, installs its `tests` directory in with my other packages. This causes problems in various cases where the packages directory that includes pyjks comes before my application's directory in `sys.path`, since my own test code directory is also called `tests`.

```
$ tar ztf pyjks-17.1.0.tar.gz
pyjks-17.1.0/
pyjks-17.1.0/tests/
pyjks-17.1.0/tests/expected/
pyjks-17.1.0/tests/expected/DSA2048.py
pyjks-17.1.0/tests/expected/RSA1024.py
pyjks-17.1.0/tests/expected/bks_custom_entry_passwords.py
pyjks-17.1.0/tests/expected/RSA2048_3certs.py
pyjks-17.1.0/tests/expected/__init__.py
pyjks-17.1.0/tests/expected/custom_entry_passwords.py
pyjks-17.1.0/tests/expected/bks_christmas.py
pyjks-17.1.0/tests/expected/jks_non_ascii_password.py
pyjks-17.1.0/tests/test_jks.py
pyjks-17.1.0/tests/__init__.py
pyjks-17.1.0/PKG-INFO
pyjks-17.1.0/setup.py
pyjks-17.1.0/pyjks.egg-info/
pyjks-17.1.0/pyjks.egg-info/PKG-INFO
pyjks-17.1.0/pyjks.egg-info/requires.txt
pyjks-17.1.0/pyjks.egg-info/dependency_links.txt
pyjks-17.1.0/pyjks.egg-info/top_level.txt
pyjks-17.1.0/pyjks.egg-info/SOURCES.txt
pyjks-17.1.0/jks/
pyjks-17.1.0/jks/rfc2898.py
pyjks-17.1.0/jks/bks.py
pyjks-17.1.0/jks/sun_crypto.py
pyjks-17.1.0/jks/util.py
pyjks-17.1.0/jks/__init__.py
pyjks-17.1.0/jks/rfc7292.py
pyjks-17.1.0/jks/jks.py
pyjks-17.1.0/setup.cfg
```

This is probably happening because `setup.py` does not use the `exclude` argument to exclude the `tests` directory/package/module:

http://setuptools.readthedocs.io/en/latest/setuptools.html#using-find-packages

**Comment by aetherknight** [comment]
Actually, I'm not sure whether the tests file should or should not be excluded from the tar archive, since end users might want to run tests (I'm not sure how setuptools handles this compared to other languages/package managers).

A better example would be to:
```
$ mkdir foo
$ pip install pyjks -t foo
Collecting pyjks
  Using cached pyjks-17.1.0-py2.py3-none-any.whl
Collecting pycryptodome (from pyjks)
Collecting twofish (from pyjks)
Collecting pyasn1-modules (from pyjks)
  Using cached pyasn1_modules-0.1.5-py2.py3-none-any.whl
Collecting pyasn1 (from pyjks)
  Using cached pyasn1-0.3.7-py2.py3-none-any.whl
Collecting javaobj-py3 (from pyjks)
  Using cached javaobj_py3-0.2.3-py2.py3-none-any.whl
Installing collected packages: pycryptodome, twofish, pyasn1, pyasn1-modules, javaobj-py3, pyjks
Successfully installed javaobj-py3-0.2.3 pyasn1-0.3.7 pyasn1-modules-0.1.5 pycryptodome-3.4.7 pyjks-17.1.0 twofish-0.3.0

$ ls foo
Crypto
_twofish.so
javaobj.py
javaobj.pyc
javaobj_py3-0.2.3.dist-info
jks
pyasn1
pyasn1-0.3.7.dist-info
pyasn1_modules
pyasn1_modules-0.1.5.dist-info
pycryptodome-3.4.7.dist-info
pyjks-17.1.0.dist-info
tests
twofish-0.3.0.dist-info
twofish.py
twofish.pyc

$ ls foo/tests
__init__.py
__init__.pyc
expected
test_jks.py
test_jks.pyc
```

**Comment by mahmoud** [comment]
Ah, yes, I can confirm this was unintentional. If we wanted to ship tests, we'd put them under the `jks` package. In any case, I'll get this excluded and reuploaded shortly. Thanks very much for the report!

**Comment by mahmoud** [comment]
Just released version 17.1.1 to PyPI, and a pip install confirms `tests` are excluded. I've added a MANIFEST.in, as well as a `tox` environment that'll check it automatically going forward. Thanks again for the in-depth report, @aetherknight!
